src/ipc/ipc4/handler.c

```diff
@@ -914,6 +914,99 @@ static int ipc4_unbind_module_instance(struct ipc4_message_request *ipc4)
 	return ipc_comp_disconnect(ipc, (ipc_pipe_comp_connect *)&bu);
 }
 
+static int  ipc4_get_vendor_config_module_instance(struct comp_dev *dev,
+						   const struct comp_driver *drv,
+						   uint32_t module_id,
+						   uint32_t instance_id,
+						   bool init_block,
+						   bool final_block,
+						   uint32_t *data_off_size,
+						   char *data_out,
+						   const char *data_in)
+{
+	const struct sof_tl *input_tl = (struct sof_tl *)data_in;
+	int ret;
+	struct vendor_error *error;
+
+	if (init_block && final_block) {
+		/* we use data_off_size as in/out,
+		 *  save value to new variable so it can be used as out size
+		 */
+		const uint8_t *end_offset = data_in + *data_off_size;
+		size_t produced_data = 0;
+
+		while ((const uint8_t *)input_tl < end_offset) {
+			/* we go to next output tlv with each iteration */
+			uint32_t data_off_size_local;
+			struct sof_tlv *output_tlv = (struct sof_tlv *)(data_out + produced_data);
+
+			if (produced_data + input_tl->max_length > MAILBOX_DSPBOX_SIZE)
+				return IPC4_FAILURE;
```

Comment:
I'm guessing ipc response finds its way to dspbox in the end, so is comparing this to dspbox size correct?

Reply:
I'll add logs in a bit.
Just wanted to be sure if data actually ends up copied to the window. Ipc code seems a bit messy so it's kinda hard to find the exact moment this happens.